service/history/api/signalwithstartworkflow/signal_with_start_workflow.go

```diff
@@ -170,13 +170,15 @@ func createVersionedRunID(currentWorkflowLease api.WorkflowLease) (*api.Versione
 		return nil, nil
 	}
 	currentExecutionState := currentWorkflowLease.GetMutableState().GetExecutionState()
-	currentLastWriteVersion, err := currentWorkflowLease.GetMutableState().GetLastWriteVersion()
+	currentCloseVersion, err := currentWorkflowLease.GetMutableState().GetCloseVersion()
```

Comment:
what if the workflow is running

Reply:
This function is only called when current workflow is closed and we are creating a new run.